packages/ra-core/src/core/useScrollToTop.tsx

```diff
@@ -23,18 +23,10 @@ import { useHistory } from 'react-router-dom';
  * );
  */
 export const useScrollToTop = () => {
-    const history = useHistory<{ _scrollToTop?: boolean }>();
-    useEffect(
-        () =>
-            history.listen((location, action) => {
-                if (
-                    action !== 'POP' &&
-                    location.state?._scrollToTop &&
-                    typeof window != 'undefined'
-                ) {
-                    window.scrollTo(0, 0);
-                }
-            }),
-        [history]
-    );
+    const location = useLocation();
+    useEffect(() => {
+        if (location.state?._scrollToTop && typeof window != 'undefined') {
+            window.scrollTo(0, 0);
+        }
+    }, [location]);
```

Comment:
this is less effective than the previous system, which only registered the listener once. Is it no longer possible?

Reply:
I don't get it, there's no listener at all. We just react to location changes